pytorch_lightning/loggers/wandb.py

```diff
@@ -124,7 +126,7 @@ def experiment(self) -> Run:
                 os.environ['WANDB_MODE'] = 'dryrun'
             self._experiment = wandb.init(
                 name=self._name, dir=self._save_dir, project=self._project, anonymous=self._anonymous,
-                reinit=True, id=self._id, resume='allow', **self._kwargs)
+                id=self._id, resume='allow', **self._kwargs) if wandb.run is None else wandb.run
```

Comment:
co this is change cor some new version? can we add a note/explanation?

Reply:
Sure I'll add in the Changelog. Basically the old wandb version forced us to use `reinit=True` to handle all the PL special cases, in particular related to distributed computing. This in turn created some limitations.
Now it should work well for every possible cases 🤞